### PR TITLE
Add metadata labels

### PR DESCRIPTION
**Pull request by timja**
Replaces https://github.com/jenkinsci/docker-agent/pull/141 which was heavily conflicted

### CODE REVIEW

**Review by MarkEWaite — APPROVED** [review]

File: tests/agent.Tests.ps1 [hunk 1]
```diff
@@ -133,6 +133,12 @@ Describe "[$global:JDK $global:FLAVOR] check user access to directories" {
         $exitCode | Should -Be 0
     }
 
+    It 'version in docker metadata' {
+        $exitCode, $stdout, $stderr = Run-Program 'docker.exe' "inspect -f `"{{index .Config.Labels \`"org.opencontainers.image.version\`"}}`" $SUT_IMAGE"
+        $exitCode | Should -Be 0
```
Comment (MarkEWaite) [inline tests/agent.Tests.ps1:138]: Is there an unbalanced grave ("`") on line 76 in the file?

Comment (timja) [reply tests/agent.Tests.ps1:138]: 76? I haven't touched around there.

File: tests/agent.Tests.ps1 [hunk 2]
```diff
@@ -133,6 +133,12 @@ Describe "[$global:JDK $global:FLAVOR] check user access to directories" {
         $exitCode | Should -Be 0
     }
 
+    It 'version in docker metadata' {
+        $exitCode, $stdout, $stderr = Run-Program 'docker.exe' "inspect -f `"{{index .Config.Labels \`"org.opencontainers.image.version\`"}}`" $SUT_IMAGE"
```
Comment (timja) [inline tests/agent.Tests.ps1:137]:
```suggestion
        $exitCode, $stdout, $stderr = Run-Program 'docker.exe' "inspect -f `"{{index .Config.Labels \`"org.opencontainers.image.version\`"}}`" $global:AGENT_IMAGE"
```

File: tests/agent.Tests.ps1 (hunk 1 above)
Comment (timja) [reply tests/agent.Tests.ps1:138]: I ran the linux tests locally, but the windows one's is just me copy pastings from jenkinsci/docker and hoping for the best, I'll start a VM if I need to

**Pull request merged**